eppoclient/httpclient.go

```diff
@@ -59,22 +66,25 @@ func (hc *httpClient) get(resource string) ([]byte, error) {
 		// error.
 		//
 		// We should almost never expect to see this condition be executed.
-		return nil, err
+		return HttpResponse{}, err // Return empty strings and the error
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode == 401 {
-		hc.isUnauthorized = true
-		return nil, fmt.Errorf("unauthorized access")
+		return HttpResponse{}, fmt.Errorf("unauthorized access") // Return an error indicating unauthorized access
 	}
 
 	if resp.StatusCode >= 500 {
-		return nil, fmt.Errorf("server error: %d", resp.StatusCode)
+		return HttpResponse{}, fmt.Errorf("server error: %d", resp.StatusCode) // Handle server errors (status code > 500)
 	}
 
 	b, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return nil, fmt.Errorf("server error: unreadable body")
+		return HttpResponse{}, fmt.Errorf("server error: unreadable body") // Return empty strings and the error
 	}
-	return b, nil
+
+	return HttpResponse{
+		Body: string(b),
+		ETag: resp.Header.Get("ETag"), // Capture the ETag header
```

Comment:
I am learning that there are two types but `ETag` (capital T) is the canonical form: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag

The Eppo Fastly CDN seems to return `Etag` prefix with a `W` which is a "weak" tag. I think this is because by default express, the web server underlying NestJS, adds this by default.

Before we release this, Eppo's web server might need to be updated to return "strong" etags.

Reply:
Weak ETag is fine for our purposes as we don't care about byte-by-byte identity—we only need the json value to match.